File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right, max_area = 0, len(height) - 1, 0

        while left < right:
            width = right - left
            cur_area = min(height[left], height[right]) * width
            max_area = max(cur_area, max_area)
            if height[left] < height[right]: left += 1
            else: right -= 1
        
        return max_area

File: 0011-container-with-most-water/0011-container-with-most-water.go
func maxArea(height []int) int {

    n := len(height)
    area := 0

    left, right := 0, n-1

    for left < right {
        width := right - left

        hei := min(height[left], height[right])

        curArea := width * hei

        area = max(curArea, area)

        if height[left] < height[right] {
            left++
        } else {
            right--
        }
    }

    return area
}